tests/MongoDB.Driver.Tests/AsyncCursorTests.cs

```diff
@@ -53,6 +56,54 @@ public void Cursor_should_not_throw_exception_after_double_close([Values(false,
             }
         }
 
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void KillCursor_should_actually_work(
+            [Values(false, true)] bool async)
+        {
+            RequireServer.Check().Supports(Feature.KillCursorsCommand);
+            var eventCapturer = new EventCapturer().Capture<CommandSucceededEvent>(x => x.CommandName.Equals("killCursors"));
+            using (var client = DriverTestConfiguration.CreateDisposableClient(eventCapturer))
+            {
+                IAsyncCursor<BsonDocument> cursor;
+                var database = client.GetDatabase("admin");
+                var collection = database.GetCollection<BsonDocument>(GetType().Name);
+                if (async)
```

Comment:
The tests will run measurably faster if you use InsertMany instead of InsertOne.

Reply:
Done